module/surveys/helper.js

```diff
@@ -1036,6 +1058,43 @@ module.exports = class SurveysHelper {
                     submissionDocumentEvidences = surveySubmissionDocument[0].evidences;
 
                 } else {
+                    // join survey's program. PII data consent is given via this api call.
+                    if ( solutionDocument.programId && userToken !== "" ) {
+                        // check if already joined the program
+                        //Check data present in programUsers collection.
+                        const programUsers = await programUsersHelper.programUsersDocuments(
+                            {
+                                userId : userId,
+                                programId : solutionDocument.programId
+                            },
+                            ["_id"]
+                        );
+                        
+                        if ( !programUsers.length > 0 && ( appVersion === "" || appVersion < 5.2 ) ) {
```

Comment:
@VISHNUDAS-tunerlabs we need to see if there is other way, we cannot use static version like this.